ai_graph_color/problem_generator.py

```diff
@@ -17,15 +18,21 @@ def write_graph_to_file(out_file, graph):
     :rtype: None
     :return: Nothing, but a file is written
     """
-    out_file = os.path.join('problems', out_file)
+    out_file = create_file(out_file)
     with open(out_file, 'w') as out_file:
-        out_file.writelines(
-            ' '.join(
-                map(
-                    lambda x: str(x), adjacency_list
-                )
-            ) + '\n' for adjacency_list in graph
-        )
+        json.dump(graph, out_file, encoding='utf8')
+
+
+def create_file(file_name):
```

Comment:
This function seems pretty vaguely named and commented; maybe something like `build_problem_path` or something would be a better name?